database/memory/db.go

```diff
@@ -375,12 +398,53 @@ func (db *MemoryDB) GetEventsFromAddressTotal(address types.Address, options *ty
 	return uint64(len(db.eventIndexDB[address])), nil
 }
 
-func (db *MemoryDB) GetStorageWithOptions(types.Address, *types.PageOptions) ([]*types.StorageResult, error) {
-	return nil, database.ErrNotImplemented
+func (db *MemoryDB) GetStorageWithOptions(address types.Address, options *types.PageOptions) ([]*types.StorageResult, error) {
+	db.mux.RLock()
+	defer db.mux.RUnlock()
+	if !db.addressIsRegistered(address) {
+		return nil, errors.New("address is not registered")
+	}
+	var convertedList []*types.StorageResult
+
+	fromBlockNum := options.BeginBlockNumber.Uint64()
+	endBlockNum := options.EndBlockNumber.Int64()
+	blockNum := fromBlockNum
+
+	storageIndexer, ok := db.storageIndexDB[address]
+	if ok {
+		for blkNum, storageRoot := range storageIndexer.root {
+			if blkNum >= fromBlockNum && (blkNum <= uint64(endBlockNum) || endBlockNum == -1) {
+				convertedList = append(convertedList, &types.StorageResult{
+					Storage:     storageIndexer.storage[storageRoot],
+					StorageRoot: types.NewHash(storageRoot),
+					BlockNumber: blockNum,
+				})
+			}
+		}
+	}
+
+	return convertedList, nil
```

Comment:
Currently, this return list isn't sorted on block number. I think it makes sense to do that to keep in line with the ES implementation.

Reply:
done

Reply:
I think this may be sorted the wrong way. In elasticsearch the results are sorted in descending order so that the most recent block/tx is shown first. In most of the in-memory results it looks like it's in ascending order.

Reply:
i have fixed it,  _GetStorageHistory_ will return results in descending order in-memory. All other apis return data in ascending order in both in-memory and elasticsearch